### PR TITLE
Fix error while running format_ninja.py on windows

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/246854012

Fixes this error: ValueError: path is on mount '\\\\.\\nul', start on mount 'C:'
More details in b/246854012#comment39

### CODE REVIEW

File: tools/format_ninja.py [hunk 1]
```diff
@@ -56,6 +56,11 @@ def remove_directory_path(path: str, directory: str) -> str:
 
 
 def normalize_if_pathlike(path: str, directory: str) -> str:
+
+  # Fixes an issue on windows where paths are parsed incorrectly.
+  if path == 'nul':
```
[the remark this answers is not in the document]
Comment (niranjanyardi) [reply tools/format_ninja.py:61]: the former. 
the argument is actually the string "nul"

File: tools/format_ninja.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def remove_directory_path(path: str, directory: str) -> str:
 
 
 def normalize_if_pathlike(path: str, directory: str) -> str:
+
```
Comment (arjungm) [inline tools/format_ninja.py:59]: nit empty line

**Review by arjungm — APPROVED** [review]
minor nit, otherwise LGTM

File: tools/format_ninja.py (hunk 2 above)
Comment (niranjanyardi) [reply tools/format_ninja.py:59]: i think the pre-commit adds this, if the pre-commit checks fails will remove this.

**Review by kaidokert — APPROVED** [review]

**Pull request merged**